### PR TITLE
[IAP][UniT] Unit Testing for Storekit Implementation Features

### PR DESCRIPTION
**Pull request by sadat-399**
- Writedown the test cases for checking different method and properties are working as expected or not.
  - All cases executed successfully.
  - Model name 'Product' need to change into 'InAppProduct'.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111

### CODE REVIEW

File: Tests/EasyPurchaseTests/FetchProductTests.swift [hunk 1]
```diff
@@ -49,8 +97,116 @@ class FetchProductTests: XCTestCase {
...
+        // Create a delay to allow time for the cancellation to take effect
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            // Fulfill the expectation to signal the completion of the test
+            expectation.fulfill()
+        }
+        // Wait for the expectation to be fulfilled (or timeout after 2 seconds)
+        wait(for: [expectation], timeout: 2.0)
+    }
+
+    func testFetchProductWithError() {
```
Comment (shamin26) [inline Tests/EasyPurchaseTests/FetchProductTests.swift:153]: Should be replaced with new one.

File: Tests/EasyPurchaseTests/FetchProductTests.swift [hunk 2]
```diff
@@ -49,8 +97,116 @@ class FetchProductTests: XCTestCase {
...
+        mockProductRequest.mockError = NSError(domain: "TestDomain", code: 123, userInfo: [NSLocalizedDescriptionKey: "An error occurred"])
+        // Set mock invalid product identifiers for the product request
+        mockProductRequest.mockInvalidProductIDs = Set([invalidProductIdentifier])
+        // Start the mock product request
+        mockProductRequest.start()
+        // Wait for the expectation to be fulfilled (or timeout after 5 seconds)
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    func testTryToFetchMultipleProducts() {
```
Comment (shamin26) [inline Tests/EasyPurchaseTests/FetchProductTests.swift:182]: Should be checked more if its working fine.

**Review by shamin26 — COMMENTED** [review]
Looks good to me. Adhoc reviewer should confirm.

**Review by shamin26 — APPROVED** [review]

**Review by TulonBJIT — APPROVED** [review]
Unable to test because this is dependant to the below PR.
"https://github.com/BJIT-RnD/EasyPurchase/pull/6"

I will merge both of these 2 branches with develop, then test and give review feedback. If there is any feedback, then we have to solve it by creating a new review ticket.

**Pull request merged**